pow/pow.go

```diff
@@ -16,9 +16,43 @@
 
 package pow
 
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type Block interface {
+	Difficulty() *big.Int
+	HashNoNonce() common.Hash
+	Nonce() uint64
+	MixDigest() common.Hash
+	NumberU64() uint64
+}
+
+type ChainManager interface {
+	GetBlockByNumber(uint64) *types.Block
+	CurrentBlock() *types.Block
+}
```

Comment:
Already down in my PoA PR which cleans up the package as a whole (also dropping Blocks). Will rather postpone to that.